### ISSUE TITLE
Ability to modify JSON schema property name

### ISSUE THREAD

**Issue by yeralin** [fuhrysteve/marshmallow-jsonschema#83]
Hi,
 
I have a problem with `marshmallow-jsonschema` where during the translation it uses field names rather that schema attributes.

For example, I have a following schema:
```
class ExampleSchema(Schema):
    api_token = String(attribute='api-token', required=True,
                       description='API token.')
```
If I run `JSONSchema().dump(ExampleSchema()).data`, it will translate to:
```
{
    "$ref": "#/definitions/LoginTokenRequestSchema",
    "$schema": "http://json-schema.org/draft-07/schema#",
    "definitions": {
        "ExampleSchema": {
            "additionalProperties": false,
            "properties": {
                "api_token": {
                    "description": "API token.",
                    "title": "api-token",
                    "type": "string"
                }
            },
            "required": [
                "api_token"
            ],
            "type": "object"
        }
    }
}
```

As you can see, it used the field name (`ExampleSchema.api_token`) from the schema instead of its attribute (`api-token`). `attribute='api-token'` was used for `title` in the resulting JSON schema.

Originally in the marshmallow core, `attribute` parameter is utilized for marshaling: https://marshmallow.readthedocs.io/en/2.x-line/quickstart.html#specifying-attribute-names

In my case, I'm just trying to follow PEP8 having `lower_case_with_underscores` variable names in the python code, but then in the resulting JSON schema I would want to follow `camelCase`. I was hoping to achieve that specifying `attribute` parameter.

Is there some kinda parameter I could pass to JSONSchema to pick schema's attribute instead?

**Comment by fuhrysteve** [comment]
Interesting.. I'm curious what your the use case is.. In other words: why do you care how the field name is spelled as long as it's both unique, descriptive, and consistent?

**Comment by yeralin** [comment]
In my case, I essentially have JSON payloads that contain an `apiToken` field:
```
{
    'apiToken': 'someToken',
    ...
}
```
I have this `ExampleSchema` for deserialization of the input payload. Now, all I am trying to do is to generate JSON Schema out of the python marshmallow classes that would conform with original input payloads.

I hope it makes sense? 😅

**Comment by yeralin** [comment]
Ok, I looked deeper into the issue. I found the following, the actual entry key `api_token` is set here:
https://github.com/fuhrysteve/marshmallow-jsonschema/blob/b29a8afd11d8eb7f0d34a8ad7de72cf102105a89/marshmallow_jsonschema/base.py#L112

Then I looked at the impl of `marshmallow.fields.field` class, and found this:
https://github.com/marshmallow-code/marshmallow/blob/e0163744069de0eacc0e70fbc0d8d5937bab753a/src/marshmallow/fields.py#L335-L343
Basically, it is not possible to change `field.name` from a client's side. It only gets modified in the internal method `_bind_to_schema`

Then, the only way is to change it in `marshmallow-jsonschema` side. How about looking for `name` extra argument under a field like:
```
class ExampleSchema(Schema):
    api_token = String(name='apiToken', required=True,
                       description='API token.')
```

Then, on `marshmallow-jsonschema` side doing something like:
```
properties[field.metadata['name'] or field.name] = schema
```
